chains/ethereum/transactions/fetchPaidAgreements.js

```diff
@@ -2,14 +2,18 @@ import { ethers } from "ethers";
 
 export const fetchPaidAgreements = async (details) => {
   try {
+    // Define the starting block
+    const startBlock = parseInt(process.env.NEXT_PUBLIC_DEPLOYMENT_BLOCK_NUMBER);
```

Comment:
I don't know if I like this approach of having block number as env variable. Is there another way to fetch blockchain data ([dGraph](https://thegraph.com/)?)